apis/v1alpha1/gateway_types.go

```diff
@@ -292,6 +294,26 @@ type Listener struct {
 	// are compatible with the application protocol specified in
 	// the Protocol field.
 	//
+	// Although a client request may technically match multiple route rules,
+	// only one may ultimately receive the request. Matching precedence MUST be
+	// determined in order of the following criteria:
+	//
+	// * The most specific match. For example, the most specific HTTPRoute match
+	//   would be determined by the longest matching combination of hostname and
+	//   path.
+	// * The oldest Route based on creation timestamp. For example, a Route with
+	//   a creation timestamp of "2020-09-08 01:02:03" would be given precedence
+	//   over a Route with a creation timestamp of "2020-09-08 01:02:04".
+	// * The Route appearing first in alphabetical order (namespace/name). For
+	//   example, foo/bar would be given precedence over foo/baz.
```

Comment:
What if multiple HTTPRouteHost within a HTTPRoute match the same hostname?

Reply:
Then the match should be based on the most specific match.

Reply:
> Then the match should be based on the most specific path/header match.

@hbagdi is this documented? If not where should this be documented? I reviewed the docs for `HTTPRouteHost` and `HTTPRouteMatch` and could not come to this conclusion.

Reply:
I asked because if the precedence cannot be determined from the most specific match within a HTTPRoute, the second and third criteria do not apply here. But I suppose within a HTTPRoute, such conflict happens rarely.

Reply:
Couple of updates:
- #337 to track @danehans's comment
- I'm not sure I completely understand you, @yiyangy. Today in our weekly sync, we discussed removing multiple hosts from HTTPRoute to simplify the resource and that might solve the problem you are trying to point out here. If you could share examples of what you mean here, that would be very helpful.

Reply:
While reviewing, I found the following in gateway_types.go:
```
	// * The most specific match. For example, the most specific HTTPRoute match
	//   is determined by the longest matching combination of hostname and path.
```

@danehans Does that help?

Reply:
I think this is going away with the latest proposal to have to make `HTTPRouteHost` `HTTPRouteSpec` (no more list of hosts per Route). That's covered by https://github.com/kubernetes-sigs/service-apis/pull/336. As long as we're OK with that approach I think we can avoid anything related to conflict resolution here.

Reply:
> That's covered by #336. As long as we're OK with that approach I think we can avoid anything related to conflict resolution here.

ack